evap/results/tools.py

```diff
@@ -42,55 +45,75 @@ def __init__(self, visible_by_contribution: Iterable[UserProfile], visible_by_de
 
 
 class RatingResult:
-    def __init__(self, question, answer_counters, additional_text_result=None):
+    def __init__(self, question, answer_counters, additional_text_result=None) -> None:
         assert question.is_rating_question
         self.question = discard_cached_related_objects(copy(question))
         self.additional_text_result = additional_text_result
-
+        self.counts = None
+        self.zipped_choices = None
+        counts = OrderedDict(
+            (value, [0, name, color, value]) for (name, color, value) in self.choices.as_name_color_value_tuples()
+        )
+        counts.pop(NO_ANSWER)
+        self.colors = tuple((color for _, _, color, _ in counts.values()))
         if answer_counters is not None:
-            counts = OrderedDict((value, 0) for value in self.choices.values if value != NO_ANSWER)
             for answer_counter in answer_counters:
-                counts[answer_counter.answer] = answer_counter.count
-            self.counts = tuple(counts.values())
-        else:
-            self.counts = None
+                counts[answer_counter.answer][0] = answer_counter.count
+            self.zipped_choices = list(counts.values())
+            self.counts = tuple((count for count, _, _, _ in counts.values()))
 
     @property
-    def choices(self):
+    def choices(self) -> Choices:
         return CHOICES[self.question.type]
 
     @property
     def count_sum(self) -> int | None:
-        if not self.is_published:
+        if not is_published(self):
             return None
         return sum(self.counts)
 
     @property
-    def minus_balance_count(self) -> float:
+    def minus_balance_count(self) -> float | None:
         assert self.question.is_bipolar_likert_question
+        if not is_published(self):
+            return None
```

Comment:
In the tests, this never happens. Should this be an assert instead? If not, why?

Reply:
The check/assert is needed because `self.counts` and `self.count_sum` is accessed. I just followed the style of the other classes. We can use the assert here instead if it's better